### ISSUE TITLE
Missing styles for table borders and properties view background

### ISSUE THREAD

**Issue by flaiker** [AObuchow/Eclipse-Spectrum-Theme#137]
Hi, I was looking for an alternative to Darkest Dark (because of it being very heavy with additional stuff I don't want and commercial licensing) and found your theme. Unfortunately there are some glaring white spots. Is there a chance these can be fixed? I know some controls are not themeable on Windows though there have been some improvements in the newer Eclipse versions (like the dark scrollbar). Darkest Dark seems to have no problems at all with these though (possibly using some more aggressive workarounds).

Environment:

- Eclipse IDE for Java Developers 2020-09 (4.17.0)
- Windows 10 2004
- [SapMachine 11.0.8 JDK](https://sap.github.io/SapMachine/)
- [ABAP Development Tools for Eclipse (ADT) 3.12.5](https://tools.hana.ondemand.com/#abap)
- Spectrum Theme 1.0.0.202009070230 installed using the marketplace
- This is a fresh Eclipse installation and workspace

### Table views have white borders

![image](https://user-images.githubusercontent.com/5270359/95675917-f0ff0700-0bba-11eb-9588-6c4cd77dbdb6.png)

![image](https://user-images.githubusercontent.com/5270359/95676081-ded19880-0bbb-11eb-9421-a414fa6d1869.png)

![image](https://user-images.githubusercontent.com/5270359/95676298-6a97f480-0bbd-11eb-8152-e86c8312acb2.png)

The default dark theme has these as well:

![image](https://user-images.githubusercontent.com/5270359/95676256-f1000680-0bbc-11eb-87f7-df4f70fd54af.png)

Darkest Dark has solved this somehow.

### Properties view has a white background

![image](https://user-images.githubusercontent.com/5270359/95675950-2b68a400-0bbb-11eb-8446-5279c3efc69f.png)

![image](https://user-images.githubusercontent.com/5270359/95676003-666ad780-0bbb-11eb-9550-64d12b5398e0.png)

With default dark theme:

![image](https://user-images.githubusercontent.com/5270359/95676250-e3e31780-0bbc-11eb-81e1-e505f279d026.png)

The code itself is also entirely not color themed but I am fairly certain that's because the ABAP Development Tools are using their own color variables...

I looked through the windows tag but have not seen these issues reported yet.

**Comment by AObuchow** [comment]
@flaiker Thank you for the wonderful bug report, this is the type of bug ticket every OSS maintainer hopes for :)


> Hi, I was looking for an alternative to Darkest Dark (because of it being very heavy with additional stuff I don't want and commercial licensing) and found your theme. 

I'm happy to be able to provide an OSS alternative to Darkest Dark/DevStyle :) 

> Unfortunately there are some glaring white spots. Is there a chance these can be fixed? 

Thankfully, my university's reading week is this week, so I'm gonna try and have this fixed this week :) I just need to do a bit more investigation.

I quickly checked out the properties view on Linux in my current Eclipse installation, and things seem to look OK:
![image](https://user-images.githubusercontent.com/10300119/95696671-0ea78d00-0c0a-11eb-9c61-a04b5809acaa.png)

I'm going to check it out on Windows shortly and see how it looks there for me.

I currently suspect the issue is either occuring on all installs of Spectrum Theme on Windows or occuring when Spectrum Theme is installed with ABAP Development Tools (on Windows specifically?). It's more likely that this is just a Windows specific issue however. 

I wanted to check out the source code for ADT but was having trouble finding it (though I checked very quickly). It's possible that ADT provides it's own additional dark theme CSS to Eclipses dark theme, as was [the case for Eclemma ](https://github.com/AObuchow/Eclipse-Spectrum-Theme/issues/79#issue-643872826). However, I don't think that's the issue here.


> ### Table views have white borders
> 
> ![image](https://user-images.githubusercontent.com/5270359/95675917-f0ff0700-0bba-11eb-9588-6c4cd77dbdb6.png)

When you say "white borders" I assume you're reffering to the white lines which divide the table's rows/columns correct?

> Darkest Dark has solved this somehow.

I wasn't aware of this and am happy you brought it to my attention. 

I recall realizing that if you create a table in SWT (which is used for Eclipse's GUI) and call `table.setLinesVisible(true);`, it adds those displeasing white lines in the table when using a dark theme on Windows. 

I suspect that the properties view calls `table.setLinesVisible(true);` and unfortunately, I don't think I can disable this through Eclipse's CSS theming. Maybe the developpers of Darkest Dark theme patched the classes responsible for table views such as the properties view to disable these lines. 

@PyvesB do you have any thoughts on the above? Maybe an upstream patch could be made to allow `table.setLinesVisible()` through Eclipse CSS? I'm not sure yet if this would be possible but it seems like an interesting idea.

> The code itself is also entirely not color themed but I am fairly certain that's because the ABAP Development Tools are using their own color variables...
> 

I imagine this is ABAP code that isin't themed correctly?

If we find the source code for ADT, I should be able to fix that for you :)

**Comment by AObuchow** [comment]
> I'm going to check it out on Windows shortly and see how it looks there for me.
> 
> I currently suspect the issue is either occuring on all installs of Spectrum Theme on Windows or occuring when Spectrum Theme is installed with ABAP Development Tools (on Windows specifically?). It's more likely that this is just a Windows specific issue however.

I just tested this out on Windows 10 and was able to reproduce your bug. I'm going to try and investigate further tomorrow and see if I can come up with a fix.

**Comment by flaiker** [comment]
Thanks for the quick response!

> I wanted to check out the source code for ADT but was having trouble finding it (though I checked very quickly). It's possible that ADT provides it's own additional dark theme CSS to Eclipses dark theme, as was the case for Eclemma . However, I don't think that's the issue here.

Hmm, I vaguely recall an ADT update where the release notes specified better dark theme support. I suppose it might be hardcoded against the standard dark theme name in the worst case scenario. As far as I know there is no source code publicly available. I only know of the public update site and some guides on how to extend it which mainly deal with the backend rest framework on the application server though.

> When you say "white borders" I assume you're reffering to the white lines which divide the table's rows/columns correct?

Exactly. For comparison:

Table views in darkest dark:

![image](https://user-images.githubusercontent.com/5270359/95706821-3cb3be00-0c58-11eb-970d-0788bef1ffa8.png)

![image](https://user-images.githubusercontent.com/5270359/95707170-17737f80-0c59-11eb-84e3-740a9b374434.png)

Properties view in darkest dark:

![image](https://user-images.githubusercontent.com/5270359/95706767-07a76b80-0c58-11eb-8501-a4c4e440fe14.png)

> I imagine this is ABAP code that isin't themed correctly?

Yes. Is there some kind of central registry where language specific color variables are mapped to global ones for style settings / themes (like "ABAP Keyword" to "Keyword") where there is currently no mapping? Or how does this work? By googling I found my own documentation on what I changed to get the colors working which I somehow forgot (see .metadata folder). https://github.com/flaiker/adt-darcula

**Comment by AObuchow** [comment]
> Hmm, I vaguely recall an ADT update where the release notes specified better dark theme support. I suppose it might be hardcoded against the standard dark theme name in the worst case scenario. As far as I know there is no source code publicly available. I only know of the public update site and some guides on how to extend it which mainly deal with the backend rest framework on the application server though.

Thankfully, since I'm able to reproduce the white on white background/foreground of the properties view on Windows, I'm almost certain that I'll be able to fix that issue.

If ADT is closed source, there's a chance we can still figure out which Eclipse preference's they are defining for their source code syntax coloring (as you asked). I'll address this below.


> ![image](https://user-images.githubusercontent.com/5270359/95706821-3cb3be00-0c58-11eb-970d-0788bef1ffa8.png)

Very interesting. I'm hopeful that with an upstream change to Eclipse, this improved look could be reproduced in Spectrum Theme. 


> > I imagine this is ABAP code that isin't themed correctly?
> 
> Yes. Is there some kind of central registry where language specific color variables are mapped to global ones for style settings / themes (like "ABAP Keyword" to "Keyword") where there is currently no mapping? Or how does this work? By googling I found my own documentation on what I changed to get the colors working which I somehow forgot (see .metadata folder). [flaiker/adt-darcula](https://github.com/flaiker/adt-darcula)

I think what you're looking for are eclipse's (internal) preferences. These can be exported using the export button in the bottom left of Eclipse's preference window.
![image](https://user-images.githubusercontent.com/10300119/95708608-bd5bc580-0c2a-11eb-8dd4-7b9beb04cbdd.png)

Then checking "Export all" and specifing a file to export to.
![image](https://user-images.githubusercontent.com/10300119/95708672-debcb180-0c2a-11eb-8c47-4453bb24d085.png)

The exported .epf is just a text file that contains preferences, including key-value pairs.

Searching for "ADT" or "adt" might help you find the syntax coloring preferences used by ADT. 
Eg.

```
[andrew]$ cat spectrum_theme_yellow_prefs.epf | grep -r -i "Spectrum"
spectrum_theme_yellow_prefs.epf:/instance/org.eclipse.ui.workbench/ColorsAndFontsPreferencePage.expandedCategories=Torg.eclipse.ui.workbenchMisc\tTcom.aobuchow.themes.spectrum.themeCategory
spectrum_theme_yellow_prefs.epf:/instance/org.eclipse.ui.workbench/ColorsAndFontsPreferencePage.selectedElement=Ccom.aobuchow.themes.spectrum.BASE_COLOR
spectrum_theme_yellow_prefs.epf:/instance/org.eclipse.ui.workbench/com.aobuchow.themes.spectrum.ACCENT_COLOR=196,160,0
spectrum_theme_yellow_prefs.epf:/instance/org.eclipse.ui.workbench/com.aobuchow.themes.spectrum.ACTIVE_FONT_COLOR=0,0,0
spectrum_theme_yellow_prefs.epf:/instance/org.eclipse.ui.workbench/com.aobuchow.themes.spectrum.BACKGROUND_COLOR=14,16,20
spectrum_theme_yellow_prefs.epf:/instance/org.eclipse.ui.workbench/com.aobuchow.themes.spectrum.BASE_COLOR=23,28,38
spectrum_theme_yellow_prefs.epf:/instance/org.eclipse.ui.workbench/spectrum.dark.theme.id.com.aobuchow.themes.spectrum.BACKGROUND_COLOR=39,41,52
spectrum_theme_yellow_prefs.epf:/instance/org.eclipse.ui.workbench/spectrum.dark.theme.id.com.aobuchow.themes.spectrum.BASE_COLOR=48,50,64
```


If you find those, I might be able to provide ADT syntax coloring (although it could break if they make changes to the preference names in future updates of ADT).

**Comment by PyvesB** [comment]
> @PyvesB do you have any thoughts on the above? Maybe an upstream patch could be made to allow `table.setLinesVisible()` through Eclipse CSS? I'm not sure yet if this would be possible but it seems like an interesting idea.

Yeah, that would be easy enough. As I had an up-to-date dev environment up and running for my Planet Themes, I drafted the implementation there directly in a branch: https://github.com/PyvesB/eclipse-planet-themes/tree/table-tree-lines-visible

Feel free to take a look and play around with it. The next step would be to either copy-paste in SpectrumTheme or submit my changes as an upstream patch (feel free to just go ahead, just mention me as one of the patch's authors). Doing it in our themes directly is the fast option and adds support for Eclipse 2020-06 and 2020-09 as well. However, doing it in upstream Eclipse allows others to use the feature, leads to less code to maintain on our side and avoids the warnings in our code bases (the classes the branch uses are not API so trigger warnings when used externally like this).

Let me know what you think! :)

**Comment by AObuchow** [comment]
> > @PyvesB do you have any thoughts on the above? Maybe an upstream patch could be made to allow `table.setLinesVisible()` through Eclipse CSS? I'm not sure yet if this would be possible but it seems like an interesting idea.
> 
> Yeah, that would be easy enough. As I had an up-to-date dev environment up and running for my Planet Themes, I drafted the implementation there directly in a branch: [https://github.com/PyvesB/eclipse-planet-themes/tree/table-tree-lines-visible](https://github.com/PyvesB/eclipse-planet-themes/tree/table-tree-lines-visible?rgh-link-date=2020-10-12T19%3A28%3A55Z)

Wow that was fast :D That looks perfect!!!

> 
> Feel free to take a look and play around with it. The next step would be to either copy-paste in SpectrumTheme or submit my changes as an upstream patch (feel free to just go ahead, just mention me as one of the patch's authors). Doing it in our themes directly is the fast option and adds support for Eclipse 2020-06 and 2020-09 as well. However, doing it in upstream Eclipse allows others to use the feature, leads to less code to maintain on our side and avoids the warnings in our code bases (the classes the branch uses are not API so trigger warnings when used externally like this).
> 
> Let me know what you think! :)

I'm +1 for getting this in upstream :) Although I do think it'd be better if you submit the patch and then I can review and merge it upstream. I can open a bug about this if you're okay with that :)

@flaiker if this fix gets merged upstream, you would have to either wait for the official December release of Eclipse or download an early release ([integration build](https://download.eclipse.org/eclipse/downloads/)) to get the white table borders removed in Spectrum theme. Do you think either of those options would be okay with you?

Otherwise I copy @PyvesB fix in Spectrum theme and remove it when the december release of Eclipse comes out.

**Comment by PyvesB** [comment]
You opening the bug and me submitting the patch works as well. 👍🏻

**Comment by AObuchow** [comment]
@PyvesB Awesome :) Here's the upstream bug for reference https://bugs.eclipse.org/bugs/show_bug.cgi?id=567821

**Comment by flaiker** [comment]
> @flaiker if this fix gets merged upstream, you would have to either wait for the official December release of Eclipse or download an early release (integration build) to get the white table borders removed in Spectrum theme. Do you think either of those options would be okay with you?

> Otherwise I copy @PyvesB fix in Spectrum theme and remove it when the december release of Eclipse comes out.

You don't have to downport the upstream fix into the theme just for me. I am fine with trying out an integration build.

> Searching for "ADT" or "adt" might help you find the syntax coloring preferences used by ADT.

Heh, unfortunately filtering by SAP or ADT still results in multiple pages of findings. Filtered by looking for color values:

```
$ cat test.epf | grep -i -E ".*(sap|adt|sqlscript).*[0-9]+,[0-9]+,[0-9]+"
/instance/com.sap.adt.tools.transformation.ui/com.sap.adt.transformation.ui.xmlAttributeEqualSign=86,165,228
/instance/com.sap.adt.tools.transformation.ui/com.sap.adt.transformation.ui.xmlAttributeValue=30,181,64
/instance/com.sap.adt.tools.transformation.ui/com.sap.adt.transformation.ui.xmlComment=128,128,128
/instance/com.sap.adt.tools.transformation.ui/com.sap.adt.transformation.ui.xmlTag=64,128,128
/instance/org.eclipse.ui.editors/adtBlockHighlightingColor=255,255,255
/instance/org.eclipse.ui.editors/adtIndirectOccurrenceIndicationColor=118,146,64
/instance/org.eclipse.ui.editors/adtOccurrenceIndicationColor=190,190,190
/instance/org.eclipse.ui.editors/adtTentativeOccurrenceIndicationColor=255,255,255
/instance/org.eclipse.ui.editors/adtWriteOccurrenceIndicationColor=255,255,255
/instance/org.eclipse.ui.editors/com.sap.adt.tools.abapsource.coverage.annotation.fullyCovered.color=23,106,8
/instance/org.eclipse.ui.editors/com.sap.adt.tools.abapsource.coverage.annotation.notCovered.color=124,28,32
/instance/org.eclipse.ui.editors/com.sap.adt.tools.abapsource.coverage.annotation.partiallyCovered.color=112,106,0
/instance/org.eclipse.ui.editors/com.sap.adt.tools.abapsource.coverage.annotation.unknown.color=140,152,161
/instance/org.eclipse.ui.editors/com.sap.adt.tools.abapsource.modification.begin.color=19,232,87
/instance/org.eclipse.ui.editors/sqlScriptDeclrColor=32,36,37
/instance/org.eclipse.ui.editors/sqlScriptOccurColor=255,255,255
/instance/org.eclipse.ui.workbench.texteditor/com.sap.adt.tools.abapsource.ui.abapMethodBackground=35,38,39
/instance/org.eclipse.ui.workbench.texteditor/com.sap.adt.tools.abapsource.ui.default=169,183,198
/instance/org.eclipse.ui.workbench.texteditor/com.sap.adt.tools.abapsource.ui.keyword=204,120,50
/instance/org.eclipse.ui.workbench.texteditor/com.sap.adt.tools.abapsource.ui.literal=106,135,89
/instance/org.eclipse.ui.workbench.texteditor/com.sap.adt.tools.abapsource.ui.literal.number=104,151,187
/instance/org.eclipse.ui.workbench.texteditor/com.sap.adt.tools.abapsource.ui.sqlScriptBackground=32,36,37
/instance/org.eclipse.ui.workbench/com.sap.adt.abap.json.boolean=143,155,163
/instance/org.eclipse.ui.workbench/com.sap.adt.abap.json.default=199,205,209
/instance/org.eclipse.ui.workbench/com.sap.adt.abap.json.error=201,0,2
/instance/org.eclipse.ui.workbench/com.sap.adt.abap.json.link=0,204,255
/instance/org.eclipse.ui.workbench/com.sap.adt.abap.json.name=197,116,48
/instance/org.eclipse.ui.workbench/com.sap.adt.abap.json.null=143,155,163
/instance/org.eclipse.ui.workbench/com.sap.adt.abap.json.number=91,233,254
/instance/org.eclipse.ui.workbench/com.sap.adt.abap.json.object=199,205,209
/instance/org.eclipse.ui.workbench/com.sap.adt.abap.json.string=29,175,62
/instance/org.eclipse.ui.workbench/com.sap.adt.datapreview.amdp.ui.readOnlyColor=11,13,14
/instance/org.eclipse.ui.workbench/com.sap.adt.debugger.highlight.foreground=0,129,235
/instance/org.eclipse.ui.workbench/com.sap.adt.featureexplorer.bgColor=55,62,67
/instance/org.eclipse.ui.workbench/com.sap.adt.featureexplorer.linkColor=85,157,248
/instance/org.eclipse.ui.workbench/com.sap.adt.featureexplorer.textColor=199,205,209
/instance/org.eclipse.ui.workbench/com.sap.adt.featureexplorer.titleBgColor=87,99,107
/instance/org.eclipse.ui.workbench/com.sap.adt.profiler.configuration.section.title=86,165,228
/instance/org.eclipse.ui.workbench/com.sap.adt.profiler.configuration.section.toggle=82,103,126
/instance/org.eclipse.ui.workbench/com.sap.adt.profiler.ui.amdp.timeline.compile=53,60,65
/instance/org.eclipse.ui.workbench/com.sap.adt.profiler.ui.amdp.timeline.execution=0,190,255
/instance/org.eclipse.ui.workbench/com.sap.adt.profiler.ui.category.ABAP=0,255,255
/instance/org.eclipse.ui.workbench/com.sap.adt.profiler.ui.category.DB=255,159,0
/instance/org.eclipse.ui.workbench/com.sap.adt.profiler.ui.category.System=44,49,53
/instance/org.eclipse.ui.workbench/com.sap.adt.quickfixview.bgColor=55,62,67
/instance/org.eclipse.ui.workbench/com.sap.adt.quickfixview.linkColor=85,157,248
/instance/org.eclipse.ui.workbench/com.sap.adt.quickfixview.quickFixNameColor=235,237,239
/instance/org.eclipse.ui.workbench/com.sap.adt.quickfixview.textColor=199,205,209
/instance/org.eclipse.ui.workbench/com.sap.adt.search.ui.writeAccess=52,46,35
/instance/org.eclipse.ui.workbench/com.sap.adt.tools.abapsource.ui.abapMethodBackground=28,31,34
/instance/org.eclipse.ui.workbench/com.sap.adt.tools.abapsource.ui.comment=115,130,140
/instance/org.eclipse.ui.workbench/com.sap.adt.tools.abapsource.ui.default=159,183,198
/instance/org.eclipse.ui.workbench/com.sap.adt.tools.abapsource.ui.enhancementPopupSelectedCodeBackground=34,39,42
/instance/org.eclipse.ui.workbench/com.sap.adt.tools.abapsource.ui.error=201,0,2
/instance/org.eclipse.ui.workbench/com.sap.adt.tools.abapsource.ui.keyword=204,120,50
/instance/org.eclipse.ui.workbench/com.sap.adt.tools.abapsource.ui.literal=98,137,83
/instance/org.eclipse.ui.workbench/com.sap.adt.tools.abapsource.ui.literal.number=93,152,191
/instance/org.eclipse.ui.workbench/com.sap.adt.tools.abapsource.ui.sqlScriptBackground=\ 83,93,101
/instance/org.eclipse.ui.workbench/com.sap.adt.tools.core.ui.color.InformationControlBackgroud=57,65,70
/instance/org.eclipse.ui.workbench/com.sap.adt.tools.core.ui.color.InformationControlForegroud=199,205,209
/instance/org.eclipse.ui.workbench/com.sap.adt.transformation.ui.xmlAttributeEqualSign=86,165,228
/instance/org.eclipse.ui.workbench/com.sap.adt.transformation.ui.xmlAttributeValue=29,175,62
/instance/org.eclipse.ui.workbench/com.sap.adt.transformation.ui.xmlComment=115,130,140
/instance/org.eclipse.ui.workbench/com.sap.adt.transformation.ui.xmlProcessingInstruction=110,124,135
/instance/org.eclipse.ui.workbench/com.sap.adt.transformation.ui.xmlTag=62,124,124
/instance/org.eclipse.ui.workbench/com.sap.adt.wda.colordefinition=88,176,88
/instance/org.eclipse.ui.workbench/com.sap.adt.wda.controller.ui.editorcolor=41,47,50
/instance/org.eclipse.ui.workbench/com.sap.adt.wda.mapping.framework.mappingline=88,118,150
/instance/org.eclipse.ui.workbench/com.sap.adt.wda.mapping.framework.selectedmappingline=199,142,0
/instance/org.eclipse.ui.workbench/com.sap.cds.ui.annotation=143,155,163
/instance/org.eclipse.ui.workbench/com.sap.cds.ui.comment=115,130,140
/instance/org.eclipse.ui.workbench/com.sap.cds.ui.default=199,205,209
/instance/org.eclipse.ui.workbench/com.sap.cds.ui.error=201,0,2
/instance/org.eclipse.ui.workbench/com.sap.cds.ui.keyword=197,116,48
/instance/org.eclipse.ui.workbench/com.sap.cds.ui.literal=29,175,62
/instance/org.eclipse.ui.workbench/com.sap.cds.ui.literal.number=91,233,254
/instance/org.eclipse.ui.workbench/com.sap.cds.ui.typeusage=199,205,209
/instance/org.eclipse.ui.workbench/com.sap.rnd.ui.annotation=143,155,163
/instance/org.eclipse.ui.workbench/com.sap.rnd.ui.comment=115,130,140
/instance/org.eclipse.ui.workbench/com.sap.rnd.ui.default=199,205,209
/instance/org.eclipse.ui.workbench/com.sap.rnd.ui.error=201,0,2
/instance/org.eclipse.ui.workbench/com.sap.rnd.ui.keyword=86,165,228
/instance/org.eclipse.ui.workbench/com.sap.rnd.ui.literal=29,175,62
/instance/org.eclipse.ui.workbench/com.sap.rnd.ui.literal.number=91,233,254
```

Surely it cannot be right that each individual theme has to implement support for every programming language for syntax highlighting? Or what does support even mean. Switching these values to something else when changing the theme under the appearance settings? How does this usually work?

**Comment by PyvesB** [comment]
Upstream Eclipse patch available: https://git.eclipse.org/r/c/platform/eclipse.platform.ui/+/170740

**Comment by AObuchow** [comment]
@BeckerWdf OK, thank you for the confirmation!

**Comment by AObuchow** [comment]
> 
> 
> Upstream Eclipse patch available: [git.eclipse.org/r/c/platform/eclipse.platform.ui/+/170740](https://git.eclipse.org/r/c/platform/eclipse.platform.ui/+/170740)

I just tested the upstream patch and it's good to merge :) 

Here's how things are looking with Spectrum Theme + @PyvesB awesome patch applied :) 
![image](https://user-images.githubusercontent.com/10300119/96196864-7d445d80-0f1e-11eb-8ec5-856f7a4fd7fb.png)

Note the problems view does not have those white lines anymore, just like DevStyle.

@flaiker I still have to fix the properties view, I will try to get that done tomorrow.

**Comment by flaiker** [comment]
Looks great!

A bit unrelated but I didn't want to open another issue just for this. Do you happen to know if you can change the minimum label width for the editor tabs? In ABAP there is no concept of namespace imports so every object has to have a unique name (system wide / globally depending on if you deliver the software / where to). This results in namespace and module prefixes for class names for example rendering the editor tab labels useless if you have several tabs open (label gets cut off at ZCL_ABAPGIT_OBJ, overflow shows the actual names):

![image](https://user-images.githubusercontent.com/5270359/96222521-b74c4880-0f8c-11eb-9a0d-943cbe95dafb.png)

I searched for eclipse minimum editor tab label width but that of course only results in tabs versus spaces discussion results :)

**Comment by AObuchow** [comment]
@flaiker I'm not sure if there's such a setting. IIRC the name used in an editor's tab is dependant on the plugin that is providing the editor. So that might be a bug to report to ADT.

**Comment by AObuchow** [comment]
> I quickly checked out the properties view on Linux in my current Eclipse installation, and things seem to look OK:
> ![image](https://user-images.githubusercontent.com/10300119/95696671-0ea78d00-0c0a-11eb-9c61-a04b5809acaa.png)

In hindsight (after I figured out a fix) I now see that using a dark GTK theme while trying to reproduce this bug on Linux wasn't very smart. I can actually reproduce this bug on Linux while using a light GTK theme.

**Comment by PyvesB** [comment]
Regarding minimum editor width for tabs, the shortening is happening deep down in the SWT stack:
https://github.com/eclipse/eclipse.platform.swt/blob/eaef41f9d39c1dc9f1f1a761bc8850bbc19df606/bundles/org.eclipse.swt/Eclipse%20SWT%20Custom%20Widgets/common/org/eclipse/swt/custom/CTabFolderRenderer.java#L1460

If you wanted to change it in Spectrum without touching SWT, you could add a CSS property handler that operates on CTabItems, and then use a tiny bit of reflection so modify the values of `shortenedText` and `shortenedTextWidth`.

**Comment by BeckerWdf** [comment]
> A bit unrelated but I didn't want to open another issue just for this. Do you happen to know if you can change the minimum label width for the editor tabs? 

Can you explain a bit more in detail. What do you want to do with the label width of the editor tabs?

**Comment by flaiker** [comment]
I think the screenshot shows it quite well. You cannot identify any of the tabs because the label is cut off before the class names differ because of prefixes. You _always_ have to hover over them or use the overflow menu to find the tab you want to switch to if you have more than x open. It would be useful if the name was always shown in full in the label at the cost of displaying fewer tabs before starting the overflow.

**Comment by BeckerWdf** [comment]
> I think the screenshot shows it quite well. You cannot identify any of the tabs because the label is cut off before the class names differ because of prefixes. You _always_ have to hover over them or use the overflow menu to find the tab you want to switch to if you have more than x open. It would be useful if the name was always shown in full in the label at the cost of displaying fewer tabs before starting the overflow. In the screenshots all labels says `ZCL_ABAPGIT_OBJ`.

I don't know any specific setting for that. But maybe this helps you:
<img width="709" alt="Screenshot 2020-11-09 at 10 50 09" src="https://user-images.githubusercontent.com/28338612/98525950-78eb3600-2279-11eb-9f08-7ec3ee0be824.png">

You can disable the display of the System-ID in the editor tab. By this you can some space.

**Comment by BeckerWdf** [comment]
> I think the screenshot shows it quite well. You cannot identify any of the tabs because the label is cut off before the class names differ because of prefixes. You _always_ have to hover over them or use the overflow menu to find the tab you want to switch to if you have more than x open. It would be useful if the name was always shown in full in the label at the cost of displaying fewer tabs before starting the overflow. In the screenshots all labels says `ZCL_ABAPGIT_OBJ`.

I think that the minimum width of the editor tabs is handled by Eclipse Platform. If you open lots of Java classes with long names you also see that the names get truncated. See:
![image](https://user-images.githubusercontent.com/28338612/98526682-5574bb00-227a-11eb-865f-6567bb087898.png)

**Comment by flaiker** [comment]
Yes, and it makes sense there because Java has namespaces while in ABAP development the namespace has to be included in the object name. In your screenshot you can still guess which tab is which because the labels aren't all identical.

If technically possible it would be a great feature for ADT to overwrite this behavior or provide an option to disable truncation for ABAP objects.
